### ISSUE TITLE
Contextual substitution Beginning does not work like kerning classes

### ISSUE THREAD

**Issue by ciqven** [johanmattssonm/birdfont#172]
**OS:** Linux x86_64
**Birdfont version:** 6.7.1 (installed via the flatpak from the website)

_tldr: Contextual substitution beginning field does not specify a character class, but an exact sequence needed. This is not intuitive (to me) and contrary to the manual. How can I specify a kerning-like class here?_ 

Greetings, I am trying to make a font for an abugida with initial and isolated vowel forms. In other words, a vowel will have three forms. Initial: Ox, combining: xO, and isolated when not preceded by a valid consonant: xaO).

I am trying to use contextual substitution to handle these vowel forms. I want a rule that converts the initial form to a combining form when preceded by a consonant, so per the manual I tried to setup the following rule:

**Ligature:** o.combining
**Beginning:** b c
**Middle:** o
**End:**

However, this does not work. In the spacing view, whenever I type `bo` or `co`, the `o` does not get substituted. Only when I type exactly `bco` does the `o` get substituted. This looks like the Beginning field does not specify a character class, but rather an exact character sequence. So I would need one rule per consonant-vowel combination.

- Is this intended to work like this? The manual (https://birdfont.org/doku/doku.php/ligatures) says it "follows the same format as the kerning classes"
- If this is the intended behavior, is there a way to simplify/automate the act of adding over a thousand (20+ consonants * 10 vowels * 5 diacritical marks) rules?

**Comment by johanmattssonm** [comment]
Yes, the beginning is a character sequence not a class.

Could mark classes be a solution?

**Comment by johanmattssonm** [comment]
Thanks I will look into that later. It is not something I have added support for in Birdfont at this time.
